src/components/checkbox/checkbox.test.tsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react'
+import { fireEvent, render, screen } from '@testing-library/react'
+import { Checkbox } from './checkbox'
+
+test('renders checkbox label correctly', () => {
+  let checked = false
+  render(
+    <Checkbox
+      label="Checkbox label"
+      checked={checked}
+      onChange={() => {
+        checked = !checked
+      }}
+    />,
+  )
+  expect(screen.getByText('Checkbox label')).toBeInTheDocument()
+})
+
+test('onChange event of Checkbox working correctly', () => {
+  let checked = false
+  render(
+    <Checkbox
+      label="Checkbox label"
+      checked={checked}
+      onChange={() => (checked = !checked)}
+    />,
+  )
+  fireEvent.click(screen.getByText('Checkbox label'))
+  expect(checked).toBe(true)
+})
+
+test('renders checkbox icon container style correctly', () => {
+  let checked = true
+  render(
+    <Checkbox
+      label="Checkbox label"
+      checked={checked}
+      onChange={() => (checked = !checked)}
+    />,
+  )
+  expect(
+    screen.getByText('Checkbox label').parentElement?.firstChild,
+  ).toHaveClass('bg-blue-500')
+})
+
+test('renders indeterminate state correctly', () => {
+  let checked: boolean | 'indeterminate' = 'indeterminate'
+  render(
+    <Checkbox
+      label="Checkbox label"
+      checked={checked}
+      onChange={() =>
+        checked === 'indeterminate'
```

Comment:
Again we shouldn't be concerned about changing the state it should be simple as

```jsx
onChange={() => {
  checked = event.target.checked
}}
```

after that you check whether is checked is changed correctly or not